tests/odbc/test-odbc-mssql.cpp

```diff
@@ -75,6 +75,34 @@ TEST_CASE("MS SQL long string", "[odbc][mssql][long]")
     );
 }
 
+TEST_CASE("MS SQL identity", "[odbc][mssql][identity]")
+{
+    soci::session sql(backEnd, connectString);
+
+    struct table_with_identity_creator : public table_creator_base
+    {
+        explicit table_with_identity_creator(soci::session& sql)
+            : table_creator_base(sql)
+        {
+            sql << "create table soci_test (id integer identity(3, 2), val integer)";
+        }
+    } table_with_identity_creator(sql);
+
+    long long id;
+    REQUIRE(sql.get_last_insert_id("soci_test", id));
+    CHECK(id == 1);
```

Comment:
I think this merits a comment because it took me a long time to realize that it's 1 here, and not 0 as I expected, just because we start with 3 and use the increment of 2, so the "pre-initial value" is 1.